packages/legend-application-marketplace/src/pages/Lakehouse/dataProduct/DataProductDataAccess.tsx

```diff
@@ -87,72 +117,224 @@ const TDSColumnDocumentationCellRenderer = (
-): React.ReactNode => {
+const TDSColumnMoreInfoCellRenderer = (props: {
+  params: DataGridCellRendererParams<V1_LakehouseAccessPoint>;
+  accessGroupState: DataProductGroupAccessState;
+}): React.ReactNode => {
+  const { params, accessGroupState } = props;
   const data = params.data;
   const store = useLegendMarketplaceBaseStore();
   const enum MoreInfoTabs {
+    COLUMNS = 'Columns',
     GRAMMAR = 'Grammar',
   }
-  const [activeTab, setActiveTab] = useState(MoreInfoTabs.GRAMMAR);
+  const [selectedTab, setSelectedTab] = useState(MoreInfoTabs.COLUMNS);
   const handleTabChange = (
     event: React.SyntheticEvent,
     newValue: MoreInfoTabs,
   ) => {
-    setActiveTab(newValue);
+    setSelectedTab(newValue);
   };
   const [accessPointGrammar, setAccessPointGrammar] =
     useState<string>('Loading ...');
+  const [accessPointRelationType, setAccessPointRelationType] = useState<
+    V1_RelationType | undefined
+  >();
+  const [loadingAccessPointDetails, setLoadingAccessPointDetails] =
+    useState<boolean>(false);
 
   useEffect(() => {
     if (!data) {
       return;
     }
 
-    const fetchGrammar = async () => {
+    const fetchAccessPointGrammar = async () => {
       try {
         const grammar = await store.engineServerClient.JSONToGrammar_lambda(
-          data.func as unknown as PlainObject<V1_RawLambda>,
+          V1_serializeRawValueSpecification(data.func),
           V1_RenderStyle.PRETTY,
         );
-
         setAccessPointGrammar(grammar);
       } catch {
         throw new Error('Error fetching access point grammar');
       }
     };
 
-    fetchGrammar().catch((error) => {
-      throw new Error(`Error fetching access point grammar: ${error.message}`);
-    });
-  }, [data, store]);
+    const fetchAccessPointRelationType = async () => {
+      try {
+        const model = accessGroupState.accessState.viewerState.isSandboxProduct
+          ? guaranteeType(
+              accessGroupState.accessState.viewerState.graphManagerState
+                .graphManager,
+              V1_PureGraphManager,
+            ).getFullGraphModelData(
+              accessGroupState.accessState.viewerState.graphManagerState.graph,
+            )
+          : new V1_PureModelContextPointer(
+              // TODO: remove as backend should handle undefined protocol input
+              new V1_Protocol(
+                V1_PureGraphManager.PURE_PROTOCOL_NAME,
+                PureClientVersion.VX_X_X,
+              ),
+              new V1_LegendSDLC(
+                accessGroupState.accessState.viewerState.project.groupId,
+                accessGroupState.accessState.viewerState.project.artifactId,
+                resolveVersion(
+                  accessGroupState.accessState.viewerState.project.versionId,
+                ),
+              ),
+            );
+        const relationTypeInput = new V1_LambdaReturnTypeInput(
+          model,
+          data.func,
+        );
+        const relationType = deserialize(
```

Comment:
lets keep track on what kind of calls we make to render this page. 
we eventually want to use the artifact generation route, to cache all these. lambda relation return type is def somethign we want to cache.